src/UXClient/Components/Slider/Slider.ts

```diff
@@ -23,10 +24,40 @@ class Slider extends Component{
 	}
 
 	Slider() {
-	}
+    }
+    
+    private getXPositionOfLabel (label: string) {
+        if (this.data.map(d => d.label).indexOf(label) != -1) {
+            return this.xScale(label);
+        }
+        // find approximate position if ascending time periods and label is a time label as well
+        if ((Utils.parseTimeInput(label) > -1) && this.isAscendingTimePeriods && this.data.length > 1) {
+            let labelMillis = Utils.parseTimeInput(label);
+            for (var i = 0; i < this.data.length; i++) {
+                if (Utils.parseTimeInput(this.data[i].label) > labelMillis) {
+                    return (this.xScale(this.data[i].label) + this.xScale(this.data[Math.max(i - 1, 0)].label)) / 2;
+                } 
+            }
+            return this.xScale(this.data[this.data.length - 1].label);
+        }
+        return 0;
+    }
+
+    private determineIfAscendingTimePeriods () {
```

Comment:
private determineIfAscendingTimePeriods () {
            let left = this.data.length > 0 ? Utils.parseTimeInput(this.data[0]) : -1;
            let isAscending = left !== -1, i = 0;
            while (isAscending && i < this.data.length - 1) {
                isAscending = left < (left = Utils.parseTimeInput(this.data[i+1]));
            }
            return isAscending;
        }